azext_edge/tests/edge/orchestration/test_clone_unit.py

```diff
@@ -0,0 +1,1628 @@
+
+    restore_client: InstanceRestore = clone_state.get_restore_client(parsed_cluster_id=parsed_cluster_id)
+    restore_client.deploy(instance_name=to_instance_name)
+    deploy_body_payload = json.loads(deploy_responses[0].calls[0].request.body)
+    request_headers = deploy_responses[0].calls[0].request.headers
+
+    assert request_headers["CommandName"] == "iot ops clone"
+    assert request_headers["x-ms-correlation-request-id"]
+    assert deploy_body_payload["properties"]["mode"] == "Incremental"
+    assert deploy_body_payload["properties"]["parameters"] == {
+        "clusterName": {"value": parsed_cluster_id["name"]},
+        "instanceName": {"value": to_instance_name},
+    }
+    assert deploy_body_payload["properties"]["template"] == deploy_body_payload["properties"]["template"]
+
+    write_to = ["my", "clone", "path"]
+    target_path = PurePath(*write_to)
+    template_content.write(target_path)
+    mock_open_write.assert_called_once_with(file=f"{target_path}.json", mode="w", encoding="utf8")
+    mock_open_write().write.assert_called_once_with(json.dumps(content, indent=2))
+
+
+@pytest.mark.parametrize(
+    "instance_version_test",
+    [
+        {"version": "1.1.50"},
+        {"version": "1.1.19"},
+        {"version": "1.0.34"},
+        {"version": "1.2.0", "error": ValidationError},
+        {"version": "2.0.0", "error": ValidationError},
+        {"version": "1.0.9", "error": ValidationError},
+        {"version": "1.2.0", "force": True},
+        {"version": "1.0.9", "force": True},
+    ],
+)
+@pytest.mark.parametrize("clone_scenario", [CloneScenario()])
+def test_clone_instance_compat(
+    mocked_cmd: Mock,
+    mocked_responses: responses,
+    clone_scenario: CloneScenario,
+    instance_version_test: dict,
+):
+    model_cluster_name = generate_random_string()
+    model_instance_name = generate_random_string()
+    model_resource_group_name = generate_random_string()
+
+    clone_scenario.bootstrap(
+        mocked_responses,
+        resource_group_name=model_resource_group_name,
+        instance_name=model_instance_name,
+        cluster_name=model_cluster_name,
+        instance_version=instance_version_test["version"],
+    )
+
+    clone = partial(
+        clone_instance,
+        cmd=mocked_cmd,
+        resource_group_name=model_resource_group_name,
+        instance_name=model_instance_name,
+        no_progress=True,
+        confirm_yes=True,
+    )
+
+    expected_error = instance_version_test.get("error")
+    force = instance_version_test.get("force")
+    if expected_error and not force:
+        mocked_responses.assert_all_requests_are_fired = False
+        with pytest.raises(ValidationError) as e:
+            clone()
+        assert "This clone client is not compatible " in str(e.value)
+        return
+
+    clone(force=force)
+
+
+LOAD_VALUE = 1000
+
+
+@pytest.mark.parametrize("add_dataflows", [200])  # total=len(dataflows)*(len(profiles)+1)
```

Comment:
nit: since these are all one value and get pulled into a single dictionary for use, it might make sense to just have the dictionary as the mark.parametrize object or even not have a mark.parametrize at all

Reply:
Disagree. Separate parameterization allows a cartesian product of variations of AIO resource configuration to validate. To achieve the same coverage via single dictionary requires a lot of dictionary variations.

Reply:
We discussed offline, and clarified the comment was specifically on single value parameterization.

Reply:
Will consider in next iteration.